src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -143,12 +122,35 @@ export class DiscoverPlugin implements Plugin<DiscoverSetup, DiscoverStart> {
       if (this.servicesInitialized) {
         return;
       }
-      const services = await buildServices(core, plugins, this.docViewsRegistry!);
+      const services = await buildServices(core, plugins);
       setServices(services);
       this.servicesInitialized = true;
     };
 
+    this.injectorEmbeddable = await this.getInjectorEmbeddable(core, plugins);
+    const docViewsRegistry = new DocViewsRegistry(this.injectorEmbeddable);
+    setDocViewsRegistry(docViewsRegistry);
```

Comment:
I save `docViewsRegistry` in services separately because of asynchronous code in `start` (line 130) and `mount` (line 96) methods, `initializeServices` function is called before `this.docViewsRegistry` can be initialized.